src/tree/tree_manager.rs

```diff
@@ -190,14 +196,17 @@ impl TreeVersion for BridgedTree {
 pub async fn extract_identity_updates<M: Middleware + 'static>(
     logs: &[Log],
     middleware: Arc<M>,
-) -> eyre::Result<BTreeMap<Root, LeafUpdates>> {
+) -> Result<BTreeMap<Root, LeafUpdates>, WorldTreeError<M>> {
     let mut tree_updates = BTreeMap::new();
 
     let mut tasks = FuturesUnordered::new();
 
     // Fetch the transactions for each log concurrently
     for log in logs {
-        let tx_hash = log.transaction_hash.expect("TODO: handle this case");
+        let tx_hash = log
+            .transaction_hash
+            .expect("Could not get transaction hash");
```

Comment:
Is this case something we need to think about? Perhaps worth using a result since we're already returning one?

Reply:
Sgtm